src/org/opendatakit/briefcase/model/FormStatus.java

```diff
@@ -25,6 +25,7 @@
   private String statusString = "";
   private final StringBuilder statusHistory = new StringBuilder();
   private boolean isSuccessful = true;
+  private int historyMaxSize = 8096;
```

Comment:
Hi @lognaturel, I'm not surprised you didn't see it jumping into this cold. The selection of 8K for the bound was arbitrary. Also, I made it a field and not a constant intentionally. If someone wanted to make it a configurable setting, it's ready to go. However, the introduction of the bound is the key to fixing the issue. The issue is O(n) allocations, n being status events, for every status update. Everyone that works on briefcase should get really acquainted with #138, #140 and #141. These are essentially the same problem in different forms. O(n) complexity in latency, storage and memory allocations, respectively. Briefcase is nearly *useless* for large data sets because of these issues.